full-moon/src/ast/parsers.rs

```diff
@@ -523,12 +542,37 @@ define_parser!(
     ParseGenericFor,
     GenericFor<'a>,
     |_, state: ParserState<'a>| {
-        let (state, for_token) = ParseSymbol(Symbol::For).parse(state)?;
-        let (state, names) = expect!(
-            state,
-            OneOrMore(ParseIdentifier, ParseSymbol(Symbol::Comma), false).parse(state),
-            "expected names"
-        );
+        let (mut state, for_token) = ParseSymbol(Symbol::For).parse(state)?;
+
+        let mut names;
+        let mut type_specifiers = Vec::new();
+
+        if cfg!(feature = "roblox") {
```

Comment:
I'm pretty sure you need `cfg_if` for this instead of `if cfg!`--otherwise, you're going to be referencing fields that don't exist, leading to a compile error. Make sure to compile this with `--no-default-features` and see if it still works.

Reply:
Hm, I just copied what was used in ParseLocalAssignment 😛, but I can look into it

Reply:
In that case, don't bother--I forgot out CI does --no-default-features, which seems to pass anyway.